crates/wast/src/wast.rs

```diff
@@ -479,6 +479,21 @@ where
         bail!("expected '{}', got '{}'", expected, actual)
     }
 
+    fn assert_exception(&mut self, result: Outcome) -> Result<()> {
+        match result {
+            Outcome::Ok(values) => bail!("expected exception, got {values:?}"),
+            Outcome::Trap(err) if err.is::<ThrownException>() => {
+                // Discard the thrown exception.
+                let _ = self
+                    .store
+                    .take_pending_exception()
+                    .expect("there should be a pending exception on the store");
```

Comment:
Reading this over, the semantics of `set_pending_exception` are such that it panics if the exception is already set, right? Would that mean that if an embedder forgot to call `take_pending_exception` and then reentered wasm after an exception, would that panic on another exception?

If not, yay! If so, I might say we should instead allow `set_pending_exception` with whatever and it just overwrites whatever was there previously.

Reply:
We did indeed have "panic on re-setting pending exception", due to [your review comment here](https://github.com/bytecodealliance/wasmtime/pull/11326#discussion_r2279374762) suggesting that we `debug_assert` to ensure no prior exception. That said, now that we have the "pending exception slot" API, this is user-exposed and clearly a different tradeoff, so I've removed that assert.